contracts/shared/ComponentService.sol

```diff
@@ -626,15 +624,33 @@ contract ComponentService is
             revert ErrorComponentServiceAlreadyRegistered(componentAddress);
         }
 
-        // check instance has assigned required role to inital owner
-        instance = _getInstance(info.parentNftId);
-        owner = info.initialOwner;
+        // check component parent nft id matches with sender nft id
+        if (info.parentNftId != senderInfo.nftId) {
+            revert ErrorComponentServiceSenderNotComponentParent(senderInfo.nftId, info.parentNftId);
+        }
+
```

Comment:
What if instance does not belong to V3 release but still wired to V3 services?
Proposal: check instance version with registry